### DISCUSSION TITLE
Trouble getting Whisper installed

### DISCUSSION THREAD

**Discussion by NAL100** [openai/whisper#906] [Q&A] [answered]
On a Macbook M1 (Apple Silicone). Put a basic python script together to run whisper but it won't work? Bolded below on what seem to be the important parts. 

It said that it couldn't install due to a conflict, then I tried to uninstall (suggested in another post) and it said it wasn't installed. So I tried to install it, it said it installed successfully, but a python script couldn't find the whisper module.

**pip3 install -U openai-whisper**
Admins-MBP:Github Admin$   Preparing metadata (setup.py) ... done
ERROR: Cannot install openai-whisper==20230117 and openai-whisper==20230124 because these package versions have conflicting dependencies.

The conflict is caused by:
    openai-whisper 20230124 depends on torch
    openai-whisper 20230117 depends on torch

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

[1]+  Exit 1                  pip3 install -U openai-whisper
Admins-MBP:Github Admin$ **pip3 uninstall openai-whisper
WARNING: Skipping openai-whisper as it is not installed.**
Admins-MBP:Github Admin$ **pip3 install -U openai-whisper**
Collecting openai-whisper
  Using cached openai-whisper-20230124.tar.gz (1.2 MB)
  Preparing metadata (setup.py) ... done
Collecting numpy
  Using cached numpy-1.24.1-cp311-cp311-macosx_11_0_arm64.whl (13.8 MB)
Collecting openai-whisper
  Using cached openai-whisper-20230117.tar.gz (1.2 MB)
  Preparing metadata (setup.py) ... done
ERROR: Cannot install openai-whisper==20230117 and openai-whisper==20230124 because these package versions have conflicting dependencies.

The conflict is caused by:
    openai-whisper 20230124 depends on torch
    openai-whisper 20230117 depends on torch

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
Admins-MBP:Github Admin$ pip install git+https://github.com/openai/whisper.git 
Defaulting to user installation because normal site-packages is not writeable
Collecting git+https://github.com/openai/whisper.git
  Cloning https://github.com/openai/whisper.git to /private/var/folders/kp/k9ky52vj2qz83h6cmd8r2fjh0000gn/T/pip-req-build-aa675xv8
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git /private/var/folders/kp/k9ky52vj2qz83h6cmd8r2fjh0000gn/T/pip-req-build-aa675xv8
  Resolved https://github.com/openai/whisper.git to commit 5c1a8c10e762bf9c29fcf6b3e40f17bc8ab09864
  Preparing metadata (setup.py) ... done
Requirement already satisfied: numpy in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from openai-whisper==20230124) (1.24.1)
Requirement already satisfied: torch in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from openai-whisper==20230124) (1.13.1)
Requirement already satisfied: tqdm in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from openai-whisper==20230124) (4.64.1)
Requirement already satisfied: more-itertools in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from openai-whisper==20230124) (9.0.0)
Requirement already satisfied: transformers>=4.19.0 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from openai-whisper==20230124) (4.26.0)
Requirement already satisfied: ffmpeg-python==0.2.0 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from openai-whisper==20230124) (0.2.0)
Requirement already satisfied: future in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from ffmpeg-python==0.2.0->openai-whisper==20230124) (0.18.3)
Requirement already satisfied: regex!=2019.12.17 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (2022.10.31)
Requirement already satisfied: pyyaml>=5.1 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (6.0)
Requirement already satisfied: packaging>=20.0 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (23.0)
Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (0.13.2)
Requirement already satisfied: filelock in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (3.9.0)
Requirement already satisfied: huggingface-hub<1.0,>=0.11.0 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (0.12.0)
Requirement already satisfied: requests in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from transformers>=4.19.0->openai-whisper==20230124) (2.28.2)
Requirement already satisfied: typing-extensions in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from torch->openai-whisper==20230124) (4.4.0)
Requirement already satisfied: charset-normalizer<4,>=2 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from requests->transformers>=4.19.0->openai-whisper==20230124) (3.0.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from requests->transformers>=4.19.0->openai-whisper==20230124) (1.26.14)
Requirement already satisfied: idna<4,>=2.5 in /Users/Admin_1/Library/Python/3.8/lib/python/site-packages (from requests->transformers>=4.19.0->openai-whisper==20230124) (3.4)
Requirement already satisfied: certifi>=2017.4.17 in /Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages (from requests->transformers>=4.19.0->openai-whisper==20230124) (2020.6.20)
Admins-MBP:Github Admin$ pip install --upgrade --no-deps --force-reinstall git+https://github.com/openai/whisper.git
Defaulting to user installation because normal site-packages is not writeable
Collecting git+https://github.com/openai/whisper.git
  Cloning https://github.com/openai/whisper.git to /private/var/folders/kp/k9ky52vj2qz83h6cmd8r2fjh0000gn/T/pip-req-build-9yhcgqo0
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git /private/var/folders/kp/k9ky52vj2qz83h6cmd8r2fjh0000gn/T/pip-req-build-9yhcgqo0
  Resolved https://github.com/openai/whisper.git to commit 5c1a8c10e762bf9c29fcf6b3e40f17bc8ab09864
  Preparing metadata (setup.py) ... done
Installing collected packages: openai-whisper
  Attempting uninstall: openai-whisper
    Found existing installation: openai-whisper 20230124
    Uninstalling openai-whisper-20230124:
      Successfully uninstalled openai-whisper-20230124
  DEPRECATION: openai-whisper is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for openai-whisper ... done
**Successfully installed openai-whisper-20230124**
Admins-MBP:Github Admin$ python3 ytaudiowhisper.py
Traceback (most recent call last):
  File "/Users/Admin_1/Github/ytaudiowhisper.py", line 3, in <module>
    import whisper
**ModuleNotFoundError: No module named 'whisper'**

**Comment by jongwook** [comment, marked as answer]
My guess is that your `pip3` and `python3` is pointing to a Python 3.11 installation, whereas `pip` is linked to a Python 3.8 installation. Try `which pip` and such to find the location. (Whisper doesn't work on Python 3.11 yet.)

Does

```bash
 /Library/Frameworks/Python.framework/Versions/3.8/bin/python ytaudiowhisper.py
```

work?

**Reply by NAL100** [reply]
Tried `which python3` and `which pip3` - both are pointing to 3.11 versions (`/Library/Frameworks/Python.framework/Versions/3.11/bin/python3` and `/pip3` respectively. The command you mentioned didn't work.

Is there a way to downgrade python and pip? I'm not interested in the latest features more just compatibility with scripts I am wanting to run.

**Reply by glangford** [reply]
Uninstall Python then reinstall - there is a 3.10.9 Python installer for Mac at python.org.

**Comment by NAL100** [comment]
[EDIT]  I seem to have gotten it working. It appears that the issue is that I had accidentally put a file named whisper.py in the local directory.

------


I may need to start a new thread for this but I am trying to get it now working on my Ubuntu VM. Whisper works fine on the command line but I want to put it in a python script. On the VM pip is 3.10 and python is 3.10.7. Any ideas?

`admin@admin1:/mnt/hgfs/Ubuntu Shared/Document_Scraping$ python3 whispertest.py
Please enter the audio file name: samplevideo.mp3
Traceback (most recent call last):
  File "/mnt/hgfs/Ubuntu Shared/Document_Scraping/whispertest.py", line 1, in <module>
    import whisper
  File "/mnt/hgfs/Ubuntu Shared/Document_Scraping/whisper.py", line 7, in <module>
    model = whisper.load_model("base")
AttributeError: partially initialized module 'whisper' has no attribute 'load_model' (most likely due to a circular import)`

The contents of whispertest.py are:
`import whisper  
print(whisper.__file__)`

Took this script from @jongwook in #143 to determine the output.

For some reason when I try to uninstall whisper it gives me this message:
`admin@admin1:/mnt/hgfs/Ubuntu Shared/Document_Scraping$ pip3 uninstall whisper
WARNING: Skipping whisper as it is not installed.
admin@admin1:/mnt/hgfs/Ubuntu Shared/Document_Scraping$ which whisper
/home/admin/.local/bin/whisper`

Note that /home/admin/.local/bin is in the PATH.

**Comment by jsjoeio** [comment]
I'm running into this on an M2.

```shell
❯ pip3 install openai-whisper

Collecting openai-whisper
  Using cached openai-whisper-20230124.tar.gz (1.2 MB)
  Preparing metadata (setup.py) ... done
Collecting numpy
  Using cached numpy-1.24.2-cp311-cp311-macosx_11_0_arm64.whl (13.8 MB)
Collecting openai-whisper
  Using cached openai-whisper-20230117.tar.gz (1.2 MB)
  Preparing metadata (setup.py) ... done
ERROR: Cannot install openai-whisper==20230117 and openai-whisper==20230124 because these package versions have conflicting dependencies.

The conflict is caused by:
    openai-whisper 20230124 depends on torch
    openai-whisper 20230117 depends on torch

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

**Comment by yeldarx** [comment]
I seem to have done everything right, but there was such a problem:

```
user$ whisper cong7.wav

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1348, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1328, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1277, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1037, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 975, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/http/client.py", line 1454, in connect
    self.sock = self._context.wrap_socket(self.sock,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 513, in wrap_socket
    return self.sslsocket_class._create(
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1071, in _create
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/ssl.py", line 1342, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.10/bin/whisper", line 8, in <module>
    sys.exit(cli())
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/transcribe.py", line 433, in cli
    model = load_model(model_name, device=device, download_root=model_dir)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/__init__.py", line 131, in load_model
    checkpoint_file = _download(_MODELS[name], download_root, in_memory)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/whisper/__init__.py", line 67, in _download
    with urllib.request.urlopen(url) as source, open(download_target, "wb") as output:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 519, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/urllib/request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:997)>

user-MacBook-Air:Downloads user$ which pip
/Library/Frameworks/Python.framework/Versions/3.10/bin/pip

user-MacBook-Air:Downloads user$ which python3
/Library/Frameworks/Python.framework/Versions/3.10/bin/python3

user-MacBook-Air:Downloads user$ which python
/usr/bin/python

```

Who can tell what the problem is and how to solve it?

**Reply by glangford** [reply]
@kyeldar When Python is installed on the Mac, there is an additional manual step required. In your Applications folder under Python 3.10, read the `ReadMe.rtf` and then run `Install Certificates.command`.
